deps/rabbit/src/rabbit_networking.erl

```diff
@@ -562,16 +577,36 @@ handshake(Ref, ProxyProtocolEnabled, BufferStrategy) ->
                 {error, protocol_error, Error} ->
                     failed_to_recv_proxy_header(Ref, Error);
                 {ok, ProxyInfo} ->
-                    {ok, Sock} = ranch:handshake(Ref),
+                    Sock = try_ranch_handshake(Ref),
                     ok = tune_buffer_size(Sock, BufferStrategy),
                     {ok, {rabbit_proxy_socket, Sock, ProxyInfo}}
             end;
         false ->
-            {ok, Sock} = ranch:handshake(Ref),
+            Sock = try_ranch_handshake(Ref),
             ok = tune_buffer_size(Sock, BufferStrategy),
             {ok, Sock}
     end.
 
+try_ranch_handshake(Ref) ->
+    try ranch:handshake(Ref) of
+        {ok, Sock} ->
+            Sock
+    catch
+        %% Don't log on Reason = closed to prevent flooding the log
+        %% specially since a TCP health check, such as the default
+        %% (with cluster-operator) readinessProbe periodically opens
+        %% and closes a connection, as mentioned in
+        %% https://github.com/rabbitmq/rabbitmq-server/pull/12304
+        exit:{shutdown, {closed, _} = Reason} ->
+            exit({shutdown, Reason});
+        exit:{shutdown, {Reason, {PeerIp, PeerPort} = PeerInfo}} ->
+            PeerAddress = io_lib:format("~ts:~tp", [rabbit_misc:ntoab(PeerIp), PeerPort]),
+            Protocol = ranch_ref_to_protocol(Ref),
+            rabbit_log:error("~p error during handshake for protocol ~p and peer ~ts",
+                             [Reason, Protocol, PeerAddress]),
+            exit({shutdown, {Reason, PeerInfo}})
```

Comment:
It's better in both catch cases to call `erlang:raise/3` so that we don't lose the stacktrace which may be propagated to other processes.